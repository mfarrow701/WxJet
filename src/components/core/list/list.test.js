import React from 'react';
import {shallow} from 'enzyme';
import List from './list';

describe('List component', () => {
    let component;

    const renderWrapper = () => {
        component = shallow(<List data={[]}/>);
    };

    beforeEach(() => {
        renderWrapper();
    });

    it('should render the list component after load', () => {
        expect(component).toMatchSnapshot();
    });

    it('should render a default list', () => {
        component.setProps({
            data: [
                {key: 'A', value: '123'},
                {key: 'B', value: '345'},
                {key: 'C', value: '567'},
                {key: 'D', value: '789'}
            ]
        });
        expect(component.find('.List-Content h6').first().text()).toEqual('A');
        expect(component.find('.List-Content p').first().text()).toEqual('123');
        expect(component.find('.List-Content h6').at(1).text()).toEqual('B');
        expect(component.find('.List-Content p').at(1).text()).toEqual('345');
        expect(component.find('.List-Content h6').at(2).text()).toEqual('C');
        expect(component.find('.List-Content p').at(2).text()).toEqual('567');
        expect(component.find('.List-Content h6').at(3).text()).toEqual('D');
        expect(component.find('.List-Content p').at(3).text()).toEqual('789');
    })
});
