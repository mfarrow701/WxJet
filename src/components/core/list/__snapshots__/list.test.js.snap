// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should render the list component after load 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    data={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "List",
    },
    "ref": null,
    "rendered": Array [],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "List",
      },
      "ref": null,
      "rendered": Array [],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
